dotcom-rendering/src/lib/DOMContentLoaded.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * @returns a Promise that resolves when the DOM is ready.
+ *
+ * @see [MDN – Checking whether loading is already complete](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event#checking_whether_loading_is_already_complete)
+ */
+export const DOMContentLoaded = new Promise<void>((resolve) => {
+	if (document.readyState === 'loading') {
+		document.addEventListener('DOMContentLoaded', () => resolve(), {
+			once: true,
+		});
+	} else {
+		resolve();
+	}
+});
```

Comment:
This is a refactor from @domlander’s work in #8343

Reply:
Yep, it looks good to me.

The [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState) claim that the document ready state will be 'interactive' _before_ the `DOMContentLoaded` event fires, so I think we're all good.